### PR TITLE
[RW-2599][risk=low] Save DUA version signed

### PR DESCRIPTION
**Pull request by andy7i**
DB, UI, API changes and tests for saving the DUA version that a user has signed.

### CODE REVIEW

File: ui/src/app/components/buttons.tsx [hunk 1]
```diff
@@ -117,19 +117,17 @@ const computeStyle = ({style = {}, hover = {}, disabledStyle = {}}, {disabled})
 
 // Set data test id = '' in the child to prevent it from propegating to all children
 export const Clickable = ({as = 'div', disabled = false, onClick = null, ...props}) => {
-  // Don't propagate test IDs to the rendered child component.
-  delete props['data-test-id'];
   return <Interactive
-    as={as} {...props}
+    // `fp.omit` used to prevent propagation of test IDs to the rendered child component.
+    as={as} {...fp.omit(['data-test-id'], props)}
```
Comment (andy7i) [inline ui/src/app/components/buttons.tsx]: I want to make a general function like `removeDataTestId(props)` to make this a bit more clear.

File: api/db/changelog/db.changelog-master.xml [hunk 2]
```diff
@@ -96,4 +96,12 @@
   <include file="changelog/db.changelog-86-drop-tos.xml"/>
   <include file="changelog/db.changelog-87-add-survey-concept-set.xml"/>
   <include file="changelog/db.changelog-88-published-workspaces.xml"/>
+  <include file="changelog/db.changelog-89-add-data-use-agreement-signed-version.xml"/>
+
+  <!--
```
Comment (gjuggler) [inline api/db/changelog/db.changelog-master.xml]: Nice addition! I really appreciate little breadcrumbs like this.

File: ui/src/app/components/buttons.tsx (hunk 1 above)
Comment (gjuggler) [reply ui/src/app/components/buttons.tsx]: That would be reasonable – but actually the fp.omit is pretty clear & concise as it stands.

File: ui/src/testing/stubs/profile-api-stub.ts [hunk 3]
```diff
@@ -111,4 +111,9 @@ export class ProfileApiStub extends ProfileApi {
       resolve({});
     });
   }
+
+  public submitDataUseAgreement(dataUseAgreementVersion: number) {
+    return Promise.resolve(this.profile);
```
Comment (gjuggler) [inline ui/src/testing/stubs/profile-api-stub.ts]: For consistency with the other stubs which actually modify the profile object, maybe you can add a line:

this.profile.dataUseAgreementSignedVersion = dataUseAgreementVersion;

**Pull request merged**